backend/src/search/views.py

```diff
@@ -22,104 +34,74 @@ async def root() -> dict:
+        file_contents = file.file.read()
+        search = SearchRepository.create_search(
+            db_session=db_session,
+            user_id=user_id,
+            file_name=file.filename,
+        )
+        task_result = celery.send_task("search", (file_contents, search.id))
+        SearchRepository.update(db_session, search, {"task_id": task_result.id})
     except Exception:
         raise HTTPException(500, detail="Internal server error.")
-
-    return {"message": "successful upload", "filename": file.filename}
+    return SearchTaskCreationResponseModel(filename=file.filename)
 
 
 @router.get(
     "/search/{search_id}/results",
     status_code=200,
     dependencies=[Depends(is_authorized)],
 )
-async def get_results(search_id: int) -> SuggestionsResponseModel:
-    result = [
-        {
-            "id": 1,
-            "name": "Wolfram Fenske",
-            "src": "DBLP",
-            "year": 2015,
-            "title": "When code smells twice as much: Metric-based detection of variability-aware code smells.",
-            "affiliation": "Otto von Guericke University of Magdeburg, Germany",
-            "venue": "International Conference on Software Engineering (ICSE)",
-        },
-        {
-            "id": 2,
-            "name": "Yang Zhang",
-            "src": "Scopus",
-            "year": 2023,
-            "title": "MIRROR: multi-objective refactoring recommendation via correlation analysis",
-            "affiliation": "Hebei University of Science and Technology",
-            "venue": None,
-        },
-        {
-            "id": 3,
-            "name": "Francesca Arcelli Fontana",
-            "src": "Google Scholar",
-            "year": 2012,
-            "title": "Evaluating the lifespan of code smells using software repository mining",
-            "affiliation": "Università degli Studi di Milano-Bicocca",
-            "venue": "IEEE/ACM International Conference on Software Engineering (ICSE)",
-        },
-        {
-            "id": 4,
-            "name": "Francesca Arcelli Fontana",
-            "src": "Google Scholar",
-            "year": 2014,
-            "title": "Evaluating the lifespan of code smells using software repository mining",
-            "affiliation": "PWr",
-            "venue": "IEEE/ACM International Conference on Software Engineering (ICSE)",
-        },
-        {
-            "id": 5,
-            "name": "John Doe",
-            "src": "Scopus",
-            "year": 2022,
-            "title": "Exploring the Future of Software Development",
-            "affiliation": "Example University",
-            "venue": None,
-        },
-    ]
-
+async def get_results(
+    search_id: int, db_session: Session = Depends(get_db_session)
+) -> SuggestionsResponseModel:
+    results = []
+    authors = AuthorRepository.find_all_by_value(db_session, "search_id", search_id)
+    for author in authors:
+        publication = PublicationRepository.find_first_by_value(
```

Comment:
We can also raise here 404 if client tries to ask for results that are still pending or resulted in error